extensions/libarchive/CompressContent.cpp

```diff
@@ -81,39 +101,42 @@ void CompressContent::initialize() {
 }
 
 void CompressContent::onSchedule(core::ProcessContext *context, core::ProcessSessionFactory *sessionFactory) {
-  std::string value;
   context->getProperty(CompressLevel.getName(), compressLevel_);
   context->getProperty(CompressMode.getName(), compressMode_);
-  context->getProperty(CompressFormat.getName(), compressFormat_);
+
+  {
+    std::string compressFormatStr;
+    context->getProperty(CompressFormat.getName(), compressFormatStr);
+    std::transform(compressFormatStr.begin(), compressFormatStr.end(), compressFormatStr.begin(), ::tolower);
+    compressFormat_ = ExtendedCompressionFormat::parse(compressFormatStr.c_str());
+    if (!compressFormat_) {
+      throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Unknown compression format: \"" + compressFormatStr + "\"");
+    }
+  }
```

Comment:
Maybe this could be extracted to a separate function as it already has its own scope.

Reply:
done, moved the checking into the extraction (getProperty) itself